core/trino-main/src/main/java/io/trino/server/PluginClassLoader.java

```diff
@@ -99,7 +99,15 @@ protected Class<?> loadClass(String name, boolean resolve)
 
             // If this is an SPI class, only check SPI class loader
             if (isSpiClass(name)) {
-                return resolveClass(spiClassLoader.loadClass(name), resolve);
+                try {
+                    return resolveClass(spiClassLoader.loadClass(name), resolve);
+                }
+                catch (ClassNotFoundException e) {
+                    if (hasClassLocally(name, resolve)) {
+                        throw new ClassNotFoundException("SPI class '%s' was not found in the SPI classloader, but was found in the '%s' plugin classloader".formatted(name, pluginName), e);
```

Comment:
This is insightful, but still not actionable.
Should the err also indicate what is the right recourse?
is it "give up, this won't ever work", "remove it from your plugin classpath", "needs to be bundled in the SPI"?

Reply:
Yeah, I've faced the issue when the class was belonging to the one of the SPI_PACKAGES but not being in the provided scope. I can add a call to action to these